src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -330,6 +328,8 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
                                     current = conv.Operand;
                                     break;
 
+                                case ConversionKind.AnonymousFunction:
+                                case ConversionKind.MethodGroup:
```

Comment:
>ConversionKind.MethodGroup [](start = 37, length = 26)

Let's investigate whether this change is going to have any impact on scenarios involving Linq expression trees

Reply:
I've added a similar test using an expression lambda. Please let me know if there's something else you were thinking of.